src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -602,37 +568,31 @@ private void OnSuggestedActionsChanged(Workspace currentWorkspace, DocumentId? c
                 }
 
                 if (currentDocumentId != workspace.GetDocumentIdInCurrentContext(buffer.AsTextContainer()) ||
-                    solutionVersion == Volatile.Read(ref _lastSolutionVersionReported))
+                    solutionVersion == Volatile.Read(ref state.Target.LastSolutionVersionReported))
                 {
                     return;
                 }
 
                 this.SuggestedActionsChanged?.Invoke(this, EventArgs.Empty);
 
-                Volatile.Write(ref _lastSolutionVersionReported, solutionVersion);
+                Volatile.Write(ref state.Target.LastSolutionVersionReported, solutionVersion);
             }
 
             public async Task<ISuggestedActionCategorySet?> GetSuggestedActionCategoriesAsync(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
             {
-                if (_workspaceStatusService != null && !await _workspaceStatusService.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false))
+                using var state = _state.TryAddReference();
+                if (state is null)
+                    return null;
+
+                if (state.Target.WorkspaceStatusService != null && !await state.Target.WorkspaceStatusService.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
i agree this maintains semantics.  however, if we don't have a WSS, we should'nt proceed eitehr (since that means we have no workspace).  but we can make those changes later.